### PR TITLE
15st1 [2716][MIG] partner_name_hide_parent

### PR DESCRIPTION
**Pull request by kanda999**
[2716](https://www.quartile.co/web?#id=2716&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

**Comment by kanda999** [comment]
We may want to put the module in the OCA as appropriate.
What do you think ? @AungKoKoLin1997

**Comment by AungKoKoLin1997** [comment]
> We may want to put the module in the OCA as appropriate. What do you think ? @AungKoKoLin1997

Okay. I will create PR for this module in OCA

File: partner_name_hide_parent/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    # Override the standard method
+    def _get_contact_name(self, partner, name):
+        # Following two lines are inserted. (QRTL)
+        if partner.hide_parent:
+            return name
+        return "{}, {}".format(
+            partner.commercial_company_name or partner.parent_id.name,
+            name,
+        )
```
Comment (yostashiro) [inline partner_name_hide_parent/models/res_partner.py]:
```suggestion
        return super()._get_contact_name(partner, name)
```

**Review by yostashiro — CHANGES_REQUESTED** [review]
Would you be able to add some explanation of in what situation this module is needed?

**Comment by yostashiro** [comment]
> Okay. I will create PR for this module in OCA

@AungKoKoLin1997 Do not rush to do this. You must first understand and be able to explain why this module is needed.

**Comment by AungKoKoLin1997** [comment]
> > Okay. I will create PR for this module in OCA
> 
> @AungKoKoLin1997 Do not rush to do this. You must first understand and be able to explain why this module is needed.

Yes. I will trace and test first.

File: partner_name_hide_parent/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+This module does the following:
+
+* Adds the ability to hide parent's name from the presentation of the partners' names according to the settings of the partners.
+* Use when you want to enter partner in account_invoice ,sale_order or others but don't want to display parent's name.
```
Comment (yostashiro) [inline partner_name_hide_parent/readme/DESCRIPTION.rst]:
```suggestion

The function provided by this module is needed when you have to set up a child partner (e.g. a branch office of a company) as an 'individual' due to how Odoo recognizes the commercial partner (i.e. if a child partner is a 'company', the partner becomes the commercial partner itself), and you do not want to show the parent's name in QWeb report or website where the partner is referenced (if the child partner is an individual, Odoo by default shows the parent's name preceding the child's name).
```

**Review by yostashiro — APPROVED** [review]
Can you please update the readme as suggested (please check and confirm the content first).

**Comment by AungKoKoLin1997** [comment]
@kanda999 
Can I start working on this module for OCA ? Is that module in stable state?

**Pull request merged**